wren-ai-service/src/web/v1/services/sql_answer.py

```diff
@@ -0,0 +1,149 @@
+import logging
+from typing import Literal, Optional
+
+from langfuse.decorators import observe
+from pydantic import BaseModel
+
+from src.core.pipeline import BasicPipeline
+from src.utils import async_timer, trace_metadata
+
+logger = logging.getLogger("wren-ai-service")
+
+
+# POST /v1/sql-answer
+class SqlAnswerRequest(BaseModel):
+    _query_id: str | None = None
+    query: str
+    sql: str
+    sql_summary: str
+    thread_id: Optional[str] = None
+
+    @property
+    def query_id(self) -> str:
+        return self._query_id
+
+    @query_id.setter
+    def query_id(self, query_id: str):
+        self._query_id = query_id
+
+
+class SqlAnswerResponse(BaseModel):
+    query_id: str
+
+
+# GET /v1/sql-answer/{query_id}/result
+class SqlAnswerResultRequest(BaseModel):
+    query_id: str
+
+
+class SqlAnswerResultResponse(BaseModel):
+    class SqlAnswerError(BaseModel):
+        code: Literal["OTHERS"]
+        message: str
+
+    status: Literal["understanding", "processing", "finished", "failed"]
+    response: Optional[str] = None
+    error: Optional[SqlAnswerError] = None
+
+
+class SqlAnswerService:
+    def __init__(
+        self,
+        pipelines: dict[str, BasicPipeline],
+    ):
+        self._pipelines = pipelines
+        self._sql_answer_results = {}
```

Comment:
I think it should be instead by the TTL cache. you could refer to the function in ask service.